save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -101,4 +162,54 @@ internal class ContestController(
             )
         }
     }
+
+    /**
+     * @param contestDto requested contest
+     * @param authentication
+     * @return [String] with response
+     */
+    @PostMapping("/create")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("permitAll()")
+    @Operation(
+        method = "POST",
+        summary = "Create a new contest.",
+        description = "Create a new contest.",
+    )
+    @Parameters(
+        Parameter(name = "contestDto", `in` = ParameterIn.DEFAULT, description = "contest requested for creation", required = true)
+    )
+    @ApiResponse(responseCode = "200", description = "Contest was successfully created.")
+    @ApiResponse(responseCode = "404", description = "Organization with given name was not found.")
+    @ApiResponse(responseCode = "403", description = "User cannot create contests with given organization.")
+    @ApiResponse(responseCode = "409", description = "Contest with given name is already present.")
+    @Suppress("TYPE_ALIAS")
+    fun createContest(
+        @RequestBody contestDto: ContestDto,
+        authentication: Authentication,
+    ): Mono<ResponseEntity<String>> = Mono.justOrEmpty(
+        Optional.ofNullable(
+            organizationService.findByName(contestDto.organizationName)
+        )
+    )
+        .switchIfEmpty {
+            Mono.error(ResponseStatusException(HttpStatus.NOT_FOUND))
+        }
+        .filter {
+            organizationPermissionEvaluator.canCreateContests(it, authentication)
+        }
+        .switchIfEmpty {
+            Mono.error(ResponseStatusException(HttpStatus.FORBIDDEN))
+        }
+        .map {
```

Comment:
Couldn't we do it in one step?
```kotlin
filter {
contestService.createContestIfNotPresent(contestDto.toContest(it))
}
```
instead of `map` + `filter`?

Reply:
That is a good idea, thanks
Will do it

Reply:
Done